ts/client/scripts/risk.ts

```diff
@@ -0,0 +1,574 @@
+    .map((a: MangoAccount) => {
+      return {
+        account: a,
+        health: a.getHealth(group, HealthType.liquidationEnd),
+        healthRatio: a.getHealthRatioUi(group, HealthType.liquidationEnd),
+        liabs: toUiDecimalsForQuote(
+          a.getLiabsValue(group, HealthType.liquidationEnd),
+        ),
+      };
+    })
+    .filter((a) => a.healthRatio < healthThresh);
+
+  const table = new Table({
+    columns: [
+      { name: 'Coin', alignment: 'right' },
+      { name: 'Oracle Price', alignment: 'right' },
+      { name: 'On-Chain Price', alignment: 'right' },
+      { name: 'Future Price', alignment: 'right' },
+      // { name: 'V4 Soft Limit', alignment: 'right' },
+      { name: 'V4 Liq Fee', alignment: 'right' },
+      { name: 'Liabs', alignment: 'right' },
+      { name: 'Liabs slippage', alignment: 'right' },
+      { name: 'Assets Sum', alignment: 'right' },
+      { name: 'Assets Slippage', alignment: 'right' },
+      // { name: 'Jup Day Volume', alignment: 'right' },
+    ],
+  });
+
+  const fileName = `/tmp/${
+    new Date().toISOString().split('T')[0]
+  }-${csvSuffix}-price_impact.csv`;
+  const csvFile = fs.createWriteStream(fileName);
+  const stream = format({ headers: true });
+  stream.pipe(csvFile);
+  stream.write([
+    'Coin',
+    'Oracle Price',
+    'On-Chain Price',
+    'Future Price',
+    // 'V4 Soft Limit',
+    'V4 Liq Fee',
+    'Liabs',
+    'Liabs slippage',
+    'Assets Sum',
+    'Assets Slippage',
+    // 'Jup Day Volume',
+  ]);
+
+  // High level solana defi stats
+  const response = await fetch('https://cache.jup.ag/stats/day');
+  const res = await response.json();
+
+  const USDC_MINT = 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v';
+  const usdcBank = group.getFirstBankByMint(new PublicKey(USDC_MINT));
+
+  // For each token
+  for (const banks of Array.from(group.banksMapByMint.values())) {
+    // TODO USDC
+    const bank = banks[0];
+
+    const onChainPrice = (
+      await (
+        await fetch(`https://price.jup.ag/v4/price?ids=${bank.mint}`)
+      ).json()
+    )['data'][bank.mint.toBase58()]['price'];
+
+    if (bank.reduceOnly > 0 || bank.tokenIndex === usdcBank.tokenIndex) {
+      continue;
+    }
+
+    // Sum of all liabs, these liabs would be acquired by liqor,
+    // who would immediately want to reduce them to 0
+    // Assuming liabs need to be bought using USDC
+    const liabs =
+      // Max liab of a particular token that would be liquidated to bring health above 0
+      mangoAccountsWithHealth.reduce((sum, a) => {
+        // How much would health increase for every unit liab moved to liqor
+        // liabprice * (liabweight - (1+fee)*assetweight)
+        const tokenLiabHealthContrib = bank.price.mul(
+          bank.initLiabWeight.sub(
+            ONE_I80F48().add(bank.liquidationFee).mul(usdcBank.initAssetWeight),
+          ),
+        );
+        // Abs liab/borrow
+        const maxTokenLiab = a.account
+          .getTokenBalance(bank)
```

Comment:
This is going to cause problems when accounts have funds on the account and on the serum oo account. 

Larger accounts often have token liabs that are offset by free funds in a oo account. I think we should sum up the actual balance and the serum free amounts. All uses of getTokenBalance() need to be evaluated.

Reply:
see change in mango account ts